mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -106,6 +106,11 @@ class LogisticRegression(override val uid: String)
       case LabeledPoint(label: Double, features: Vector) => (label, features)
     }
     val handlePersistence = dataset.rdd.getStorageLevel == StorageLevel.NONE
+    trainOnInstances(instances, handlePersistence)
+  }
+
+  protected[spark] def trainOnInstances(instances: RDD[(Double, Vector)],
```

Comment:
I don't think thats whats going on under the hood in `ALS.scala` - its passing in an RDD of `NewALS.Rating` case classes, which is the datatype that `NewALS'` train function works on (although I'm never completely sure with implicits). If I simply try and pass the RDD of LabeledPoints its a compile error (although I could be missing one of the implicit imports but I'm not sure which one).